src/System.Security.Cryptography.X509Certificates/src/System/Security/Cryptography/X509Certificates/X509CertificateCollection.cs

```diff
@@ -153,106 +97,28 @@ public override int GetHashCode()
-            }
-        }
-
-        public void RemoveAt(int index)
-        {
-            _list.RemoveAt(index);
-        }
-
-        void ICollection.CopyTo(Array array, int index)
-        {
-            NonGenericList.CopyTo(array, index);
-        }
-
-        int IList.Add(object value)
-        {
-            if (value == null)
-                throw new ArgumentNullException(nameof(value));
-
-            return NonGenericList.Add(value);
-        }
-
-        bool IList.Contains(object value)
-        {
-            return NonGenericList.Contains(value);
-        }
-
-        int IList.IndexOf(object value)
-        {
-            return NonGenericList.IndexOf(value);
-        }
-
-        void IList.Insert(int index, object value)
-        {
-            if (value == null)
-                throw new ArgumentNullException(nameof(value));
-
-            NonGenericList.Insert(index, value);
-        }
-
-        void IList.Remove(object value)
-        {
-            if (value == null)
-                throw new ArgumentNullException(nameof(value));
-
-            // On full framework this method throws when removing an item that
-            // is not present in the collection, and that behavior needs to be
-            // preserved.
-            //
-            // Since that behavior is not provided by the IList.Remove exposed
-            // via the NonGenericList property, this method can't just defer
-            // like the rest of the IList explicit implementations do.
-            //
-            // The List<T> which backs this collection will guard against any
-            // objects which are not X509Certificiate-or-derived, and we've
-            // already checked whether value itself was null.  Therefore we
-            // know that any (value as X509Certificate) which becomes null
-            // could not have been in our collection, and when not null we
-            // have a rich object reference and can defer to the other Remove
-            // method on this class.
-
-            X509Certificate cert = value as X509Certificate;
-
-            if (cert == null)
-            {
-                throw new ArgumentException(SR.Arg_RemoveArgNotFound);
-            }
-
-            Remove(cert);
+            List.Remove(value);
         }
 
-        private IList NonGenericList
+        // Although this class for compatibility must derive from CollectionBase
+        // we can change the behavior a bit to verify the type of object before
+        // we add it to the collection
+        protected override void OnValidate(Object value)
         {
-            get { return _list; }
-        }
+            base.OnValidate(value);
 
-        internal void GetEnumerator(out List<X509Certificate>.Enumerator enumerator)
-        {
-            enumerator = _list.GetEnumerator();
+            if (!(value is X509Certificate))
+              throw new ArgumentException("value");
```

Comment:
This `ArgumentException` constructor is actually for the message, not the param name. `ArgumentException(string, string)` allows specifying the param name in the second parameter, in which case `nameof` should be used. Not sure what the message should be.